File: public/js/blog/blogpost.mjs
export default class BlogPost {
    constructor(title, author, date, post) {
        this.title = title;
        this.author = author;
        this.date = date;
        this.post = post;
        this.postNumber = 0;
        this.tags = '';
        this.imagePath = '';
        this.location = '/blog/post';
    }

    printAll() {
        console.log('Title', this.title);
        console.log('Author', this.author);
        console.log('Date', this.date);
    }
}

File: src/main/java/dev/christopherbell/azurras/models/blog/BlogPost.java
package dev.christopherbell.azurras.models.blog;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.io.Serializable;
import java.util.Date;

@Getter
@AllArgsConstructor
public class BlogPost implements Serializable {
    private String author;
    private String contentText;
    private Date creationDate;
    private String description;
    private int id;
    private String imagePath;
    private String tags;
    private String title;

    public BlogPost(String author,
            String contentText,
            String description,
            String imagePath,
            String tags,
            String title) {
        this.author = author;
        this.contentText = contentText;
        this.description = description;
        this.imagePath = imagePath;
        this.tags = tags;
        this.title = title;
    }
}
